x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java

```diff
@@ -226,5 +250,24 @@ public void validate(Collection<Phase> phases) {
                 "] action may not be used in the [" + HOT_PHASE +
                 "] phase without an accompanying [" + RolloverAction.NAME + "] action");
         }
+
+        // look for phases that have the migrate action enabled and also specify allocation rules via the AllocateAction
+        String phasesWithConflictingMigrationActions = phases.stream()
+            .filter(phase -> phase.getActions().containsKey(MigrateAction.NAME) &&
+                ((MigrateAction) phase.getActions().get(MigrateAction.NAME)).isEnabled() &&
+                phase.getActions().containsKey(AllocateAction.NAME) &&
+                definesAllocationRules((AllocateAction) phase.getActions().get(AllocateAction.NAME))
+            )
+            .map(Phase::getName)
+            .collect(Collectors.joining(","));
+        if (Strings.hasText(phasesWithConflictingMigrationActions)) {
+            throw new IllegalArgumentException("phases [" + phasesWithConflictingMigrationActions + "] specify an enabled " +
+                MigrateAction.NAME + " action and the " + AllocateAction.NAME + " action. specify only one data migration action in these" +
```

Comment:
I think rather than:

```
phases [warm,cold] specify an enabled migrate action and the allocate action. specify only one data migration action in these phases
```

We should clarify that it's the actual allocation rules that cause problems (they're free to adjust replica count as much as they'd like):

```
phase [warm,cold] specifies an enabled migrate action and an allocate action with allocation rules, specify only a single data migration in each phase
```